cmd/eval.go

```diff
@@ -88,3 +36,52 @@ func validateFile(file []byte) error {
 	var holder interface{}
 	return yaml.Unmarshal(file, &holder)
 }
+
+var (
+	evalCmd = &cobra.Command{
+		Use:   "eval [file]",
+		Short: "Evaluate the combinations for a file at the given path",
+		Long: `
+Evaluate the combinations for a file at the given path. The file provided must be valid YAML.
+
+Note: the combo binary requires the --replacement flag to be explicitly set.
+
+The replacements flag allows users to specify a series of key value pairs in the form of KEY=VALUES.
+
+Example: combo eval -r REPLACE_ME=1,2,3 path/to/file
+	`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			replacements, _ := cmd.Flags().GetStringToString("replacements")
+
+			file, err := os.ReadFile(args[0])
+			if err != nil {
+				return err
+			}
+
+			if err := validateFile(file); err != nil {
+				return fmt.Errorf("failed to validate file specified: %w", err)
+			}
+
+			combinations := combination.NewStream(
+				combination.WithArgs(formatReplacements(replacements)),
+				combination.WithSolveAhead(),
+			)
+
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			generatedFile, err := generator.Evaluate(ctx, string(file), combinations)
```

Comment:
This is out-of-scope for this PR, but the usage of `string(file)` here seems like it warrants some sort of refactoring to be agnostic to the input (maybe an io.Reader?). Diving into that Evaluate function, it looks like that logic for handling multi-document YAML manifests could be deflated as an options parameter to the NewStream constructor? Note: I hadn't dived too deep into the actual codebase, so I'm likely missing a good chunk of context, but something that I had noticed that we might be able to improve in the future.

Reply:
You're right that the `string(file)` here is probably an indicator that the interface for that function should be updated. I could see a situation where we use the `io.Reader` instead to get around that wonkiness but yeah lets address that in the future.